### PR TITLE
GraalVM native support for the OpenTelemetry annotations

### PR DESCRIPTION
**Pull request by jeanbisutti**
Fixes https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/6620

### CODE REVIEW

**Review by zeitlinger — APPROVED** [review]

File: smoke-tests-otel-starter/spring-boot-common/src/main/java/io/opentelemetry/spring/smoketest/AbstractOtelSpringStarterSmokeTest.java [hunk 1]
```diff
@@ -236,8 +237,8 @@ void restTemplate() {
             traceAssert.hasSpansSatisfyingExactly(
                 span -> assertClientSpan(span, "/ping"),
                 span ->
-                    span.hasKind(SpanKind.SERVER)
-                        .hasAttribute(HttpAttributes.HTTP_ROUTE, "/ping")));
+                    span.hasKind(SpanKind.SERVER).hasAttribute(HttpAttributes.HTTP_ROUTE, "/ping"),
+                span1 -> withSpanAssert(span1)));
```
Comment (trask) [inline smoke-tests-otel-starter/spring-boot-common/src/main/java/io/opentelemetry/spring/smoketest/AbstractOtelSpringStarterSmokeTest.java:241]:
```suggestion
                span -> withSpanAssert(span)));
```

File: smoke-tests-otel-starter/spring-boot-3.2/build.gradle.kts [hunk 2]
```diff
@@ -17,6 +17,7 @@ dependencies {
   implementation("org.apache.commons:commons-dbcp2")
   implementation("org.springframework.kafka:spring-kafka")
   implementation("org.springframework.boot:spring-boot-starter-data-mongodb")
+  implementation("org.springframework.boot:spring-boot-starter-aop")
```
Comment (trask) [inline smoke-tests-otel-starter/spring-boot-3.2/build.gradle.kts:20]: does it make sense to include this in our starter somehow(?) so this core feature works without users needing to remember to add this extra dependency?

**Review by trask — APPROVED** [review]

File: smoke-tests-otel-starter/spring-boot-3.2/build.gradle.kts (hunk 2 above)
Comment (zeitlinger) [reply smoke-tests-otel-starter/spring-boot-3.2/build.gradle.kts:20]: great catch - I created https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/11775 to fix this

Comment (jeanbisutti) [reply smoke-tests-otel-starter/spring-boot-3.2/build.gradle.kts:20]: As a user,  I would be surprised if the starter added a Spring Boot dependency.

Comment (zeitlinger) [reply smoke-tests-otel-starter/spring-boot-3.2/build.gradle.kts:20]: starters should be a one-stop solution, where everything you typically need is included, including other starters

users can exclude dependencies or use the autoconfigure module if the starter doesn't fit

so the question is if the annotation is a typical use of the starter

Comment (trask) [reply smoke-tests-otel-starter/spring-boot-3.2/build.gradle.kts:20]: > so the question is if the annotation is a typical use of the starter

I think the answer to this is yes

but I don't know what the solution is (if any)

> As a user, I would be surprised if the starter added a Spring Boot dependency.

@jeanbisutti why is this surprising? thanks

Comment (jeanbisutti) [reply smoke-tests-otel-starter/spring-boot-3.2/build.gradle.kts:20]: @zeitlinger @trask The goal of this PR is to make the OTel annotations work in native mode for the OTel Spring starter. The spring-boot-starter-aop change is not related to this PR. So, I propose to mark this conversation as resolved in this PR and reply in #11775.

Comment (jeanbisutti) [reply smoke-tests-otel-starter/spring-boot-3.2/build.gradle.kts:20]: https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/11775#discussion_r1672309913

**Comment by laurit** [comment]
@jeanbisutti please rebase or merge main to fix the build

**Pull request merged**